services/user/Evaluations/plugin/src/helpers.rs

```diff
@@ -0,0 +1,216 @@
+}
+
+pub fn current_user() -> Result<AccountNumber, Error> {
+    let current_user = bindings::accounts::plugin::api::get_current_user()?;
+    let current_user = current_user.ok_or(ErrorType::NotLoggedIn)?;
+    let account_number =
+        AccountNumber::from_exact(&current_user).map_err(|_| ErrorType::InvalidAccountNumber)?;
+    Ok(account_number)
+}
+
+pub fn create_new_symmetric_key(
+    evaluation_owner: AccountNumber,
+    evaluation_id: u32,
+    users: &types::GetGroupUsers,
+    creator: AccountNumber,
+) -> Result<key_table::SymmetricKey, Error> {
+    let sorted_users = sort_users_by_account(users)?;
+    let user_settings =
+        fetch_user_settings(sorted_users.iter().map(|u| u.user.clone()).collect())?.get_user_settings;
+
+    let creator_public_key = user_settings
+        .iter()
+        .find_map(|opt_s| {
+            opt_s.as_ref().and_then(|s| {
+                if s.user == creator.to_string() {
+                    Some(s.key.clone())
+                } else {
+                    None
+                }
+            })
+        })
+        .ok_or(ErrorType::UserSettingNotFound)?;
+
+    let symmetric_key = key_table::SymmetricKey::generate(creator_public_key);
+
+    let payloads: Vec<Vec<u8>> = sorted_users
+        .iter()
+        .map(|user| {
+            let setting = user_settings
+                .iter()
+                .find(|s| s.as_ref().map(|s| s.user == user.user).unwrap_or(false))
+                .and_then(|s| s.as_ref())
+                .ok_or(ErrorType::UserSettingNotFound)?;
+            encrypt(&setting.key, &symmetric_key.key).map_err(|_| ErrorType::EncryptionFailed)
+        })
+        .collect::<Result<_, _>>()?;
+
+    let hash = symmetric_key.hash();
+
+    let packed_args = evaluations::action_structs::group_key {
+        owner: evaluation_owner,
+        evaluation_id,
+        keys: payloads,
+        hash,
+    }
+    .packed();
+    add_action_to_transaction(evaluations::action_structs::group_key::ACTION_NAME, &packed_args)?;
+    Ok(symmetric_key)
+}
+
+pub fn get_decrypted_proposals(
+    evaluation_owner: AccountNumber,
+    evaluation_id: u32,
+    group_number: u32,
+    sender: AccountNumber,
+) -> Result<Vec<(AccountNumber, Option<Vec<u8>>)>, Error> {
+    let res = fetch_and_decode(evaluation_owner, evaluation_id, group_number)?;
+    let mut proposals = res.get_group_users.ok_or(ErrorType::UsersNotFound)?;
+
+    proposals.sort_by(|a, b| {
+        parse_account_number(&a.user)
+            .map(|a| a.value)
+            .unwrap_or(0)
+            .cmp(&parse_account_number(&b.user).map(|b| b.value).unwrap_or(0))
+    });
+
+    let symmetric_key = get_symmetric_key(evaluation_owner, evaluation_id, group_number, sender)?;
+
+    let proposals = proposals
+        .iter()
+        .map(|s| {
+            let user =
+                parse_account_number(&s.user).map_err(|_| ErrorType::InvalidAccountNumber)?;
```

Comment:
You can probably just `unwrap();` the result of `parse_account_number` and `decrypt`. They are unrecoverable errors that cannot happen with normal plugin usage.